src/wwwroot/controllers/DomainManagerCtrl.js

```diff
@@ -40,14 +40,20 @@
           loadUserDomains();
         });
       }
-
       vm.activateDomain = function(domain) {
         settings.createOrActiveDomain(domain.name)
         .then(function() {
           domain.disabled = false;
         });
       }
 
+      vm.deleteDomain = function(domain, index){
+        settings.deleteDomain(domain)
+        .then(function() {
+          vm.domains.splice(index, 1);
```

Comment:
What do you think about to receive a domain object (in place of domainName and index) and remove the object from `vm.domains` array? I think that it is safe because if there ar multiple deletions simultaneously the index could be wrong